spec/services/arcgis_api/geocoder_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe ArcgisApi::Geocoder do
+  include ArcgisApiHelper
+
+  let(:subject) { ArcgisApi::Geocoder.new }
+
+  describe '#suggest' do
+    it 'returns suggestions' do
+      stub_request_suggestions
+
+      suggestions = subject.suggest('100 Main')
+
+      expect(suggestions[0][:magicKey]).to be_present
+      expect(suggestions[0][:text]).to be_present
```

Comment:
1. I personally prefer `.first` to `[0]` to avoid hardcoding numbers
2. our codebase is almost entirely lower_snake_case, let's have the Geocoder wrap the responses in a simple struct so we can access them like this:

```suggestion
      expect(suggestions.first.magic_key).to be_present
      expect(suggestions.first.text).to be_present
```